internal/controller/ratelimit_workflow_helpers.go

```diff
@@ -284,3 +261,223 @@ func rateLimitPolicyAcceptedStatus(policy machinery.Policy) (accepted bool, err
 	}
 	return
 }
+
+// TokenLimitNameToLimitadorIdentifier converts a token rate limit policy and limit name to a unique Limitador ident
+func TokenLimitNameToLimitadorIdentifier(trlpKey k8stypes.NamespacedName, uniqueLimitName string) string {
+	identifier := "tokenlimit."
+
+	for _, c := range uniqueLimitName {
+		if unicode.IsLetter(c) || unicode.IsDigit(c) || c == '_' {
+			identifier += string(c)
+		} else {
+			identifier += "_"
+		}
+	}
+
+	hash := sha256.Sum256([]byte(fmt.Sprintf("%s/%s", trlpKey.String(), uniqueLimitName)))
+	identifier += "__" + hex.EncodeToString(hash[:4])
+
+	return identifier
+}
+
+func wasmActionsFromTokenLimit(tokenLimit *kuadrantv1alpha1.TokenLimit, limitIdentifier, scope string, topLevelPredicates kuadrantv1.WhenPredicates) []wasm.Action {
+	predicates := make([]string, 0, len(topLevelPredicates)+1)
+	for _, pred := range topLevelPredicates {
+		predicates = append(predicates, pred.Predicate)
+	}
+	for _, pred := range tokenLimit.When {
+		predicates = append(predicates, pred.Predicate)
+	}
+
+	// common both actions
+	commonData := []wasm.DataType{
+		{
+			Value: &wasm.Expression{
+				ExpressionItem: wasm.ExpressionItem{
+					Key:   limitIdentifier,
+					Value: "1",
+				},
+			},
+		},
+	}
+
+	// add model detection if needed
+	modelDetectionNeeded := false
```

Comment:
I do not get this. What is the reasoning behind adding a descriptor entry for the model when `requestBodyJSON(\"model\")` is in the predicate?

Reply:
fixed for real this time